src/Core/Managed/Shared/Extensibility/Implementation/TaskTimer.cs

```diff
@@ -17,15 +17,22 @@ namespace Microsoft.ApplicationInsights.Extensibility.Implementation
     /// <summary>
     /// Runs a task after a certain delay and log any error.
     /// </summary>
-    public class TaskTimer : IDisposable
+    public class TaskTimer
```

Comment:
I wonder - do we need to make this class internal and mark the public version as Obsolete for now? I do not understand why Application Insights library should provide this functionality